sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/CollectionConfiguration.cs

```diff
@@ -54,24 +58,41 @@ internal class CollectionConfiguration
             // this includes both telemetry metrics and Metric metrics
             this.CreateMetadata();
 
-            errors = metricErrors.ToArray();
+            // create document streams based on description in info
+            this.CreateDocumentStreams(previousDocumentStreams ?? Array.Empty<DocumentStream>(), out CollectionConfigurationError[] documentStreamErrors);
+
+            errors = metricErrors.Concat(documentStreamErrors).ToArray();
 
-            foreach (var error in errors)
+            UpdateAllErrorsWithKeyValue(errors, "ETag", this.info.Etag);
+        }
+
+        private void UpdateAllErrorsWithKeyValue(CollectionConfigurationError[] errors, string key, string value)
+        {
+            for (int i = 0; i < errors.Length; i++)
             {
-                UpdateMetricIdOfError(error, this.info.Etag);
+                var newError = UpdateOrCreateError(errors[i], key, value);
+                if (newError != null)
+                {
+                    errors[i] = newError;
+                }
             }
         }
 
-        private void UpdateMetricIdOfError(CollectionConfigurationError error, string id)
+        private CollectionConfigurationError? UpdateOrCreateError(CollectionConfigurationError error, string key, string value)
         {
             for (int i = 0; i < error.Data.Count; i++)
             {
-                if (error.Data[i].Key == "ETag")
+                if (error.Data[i].Key == key)
                 {
-                    error.Data[i].Value = id;
-                    return;
+                    error.Data[i].Value = value;
+                    return null;
                 }
             }
```

Comment:
I see that you removed the hardcoded "ETag" key. 
For my information, do other kvp get updated other than etag?